src/core/symbology/qgsfillsymbollayer.cpp

```diff
@@ -3906,6 +3906,21 @@ void QgsRasterFillSymbolLayer::applyPattern( QBrush &brush, const QString &image
   if ( width > 0 )
   {
     size.setWidth( context.renderContext().convertToPainterUnits( width, mWidthUnit, mWidthMapUnitScale ) );
+
+    // RenderPercentage Unit Type takes original image size
+    if ( mWidthUnit == QgsUnitTypes::RenderPercentage )
```

Comment:
this should be in an "else" block, avoiding the unused setWidth call from above. E.g.

    if ( mWidthUnit != QgsUnitTypes::RenderPercentage )
    {
      size.setWidth( context.renderContext().convertToPainterUnits( width, mWidthUnit, mWidthMapUnitScale ) );
    }
    else
    {
       QSize originalSize = QgsApplication::imageCache()->originalSize( imageFilePath );
       ...
    }

Reply:
Did it.